### ISSUE TITLE
bug: After plugin is closed, there are still some requests to execute the plugin

### ISSUE THREAD

**Issue by whioue** [apache/apisix#7684]
### Current Behavior

当我开启自定义插件使用过一段时间后，关闭此插件，偶现部分请求仍然会执行此插件

### Expected Behavior

预期在关闭插件后不会所有请求都不会再执行此插件

### Error Logs

暂无

### Steps to Reproduce

1.编辑自定义插件，并能正常运行。
2.配置一条路由，并在路由所在的服务中开启此插件。
3.并发请求此路由，所有请求会执行此插件，并能生效。
4.关闭此插件。
5.请求此路由发现偶现个别请求仍然会执行插件功能。
当我特意去复现时我不能做到百分百复现，但此问题确实存在。考虑到是因为openresty中部分worker里的连接句柄没有得到释放导致？不知道大佬有没有遇到相关问题，以及解决的指导建议呢？

### Environment

- APISIX version (run `apisix version`): apisix v2.0
- Operating system (run `uname -a`): centos7
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`): 1.17.8.2
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

**Comment by tzssangglass** [comment]
> 考虑到是因为openresty中部分worker里的连接句柄没有得到释放导致？

I don't think it has anything to do with the file handler, the plugin information is in memory and not associated with the file handler.

for step 4, how do you close your custom plugin?

**Comment by tokers** [comment]
Check out if there are any suspicious error logs when you close the plugin and make sure the close logic is effective.

**Comment by whioue** [comment]
> > 考虑到是因为openresty中部分工人的连接句柄没有释放导致吗？
> 
> 我认为它与文件处理程序没有任何关系，插件信息在内存中并且与文件处理程序无关。
> 
> 对于第 4 步，如何关闭自定义插件？
我通过调用apisix中的服务接口（插件配置在服务上）put方法修改服务中插件，进行关闭，是否存在某worker中内存没有及时清理造成的这种情况呢

**Comment by tzssangglass** [comment]
> 我通过调用apisix中的服务接口（插件配置在服务上）put方法修改服务中插件，进行关闭，是否存在某worker中内存没有及时清理造成的这种情况呢

The worker writes the updates to etcd, and other workers watch for etcd changes and update the data in memory. There is a delay in between, but if everything is working properly, the delay should be low enough to be negligible.

**Comment by tzssangglass** [comment]
> 当我特意去复现时我不能做到百分百复现

Since this issue is related to your custom plugin, I'm going to turn this issue into a discussion. if the plugin in the official repository also has this issue, please open a separate issue to report it.
